hilti/runtime/src/types/time.cc

```diff
@@ -18,6 +18,24 @@ Time time::current_time() {
     return Time(t, Time::SecondTag());
 }
 
+Time time::mktime(uint64_t y, uint64_t m, uint64_t d, uint64_t H, uint64_t M, uint64_t S) {
+    struct tm t;
+    t.tm_sec = S;
+    t.tm_min = M;
+    t.tm_hour = H;
+    t.tm_mday = d;
+    t.tm_mon = m - 1;
+    t.tm_year = y - 1900;
```

Comment:
More an observation: The epoch will depend on whether we use POSIX or not. Currently that seems to be the case, but might become an issue one we need to support e.g., win platforms. I think above unit test might catch when the epoch becomes a problem.

Reply:
What's not portable in this part?

Reply:
You are right, `mktime` is POSIX and uses the UNIX epoch as well, https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/mktime-mktime32-mktime64?view=msvc-160.